eternalcore-core/src/main/java/com/eternalcode/core/util/CompletableFutureUtil.java

```diff
@@ -0,0 +1,18 @@
+package com.eternalcode.core.util;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public final class CompletableFutureUtil {
+
+    private static final Logger LOGGER = Logger.getLogger(CompletableFutureUtil.class.getName());
+
+    private CompletableFutureUtil() {
+        throw new UnsupportedOperationException("Utility class");
+    }
+
+    public static <T> T handleCaughtException(Throwable cause) {
```

Comment:
From what I see we don't need generic parameters here, as they aren't used in the rest of the code.